src/Bicep.Core/Syntax/ProviderDeclarationSyntax.cs

```diff
@@ -33,20 +36,40 @@ public ProviderDeclarationSyntax(IEnumerable<SyntaxBase> leadingNodes, Token key
 
         public SyntaxBase AsClause { get; }
 
-        public ImportSpecification Specification => lazySpecification.Value;
+        public IProviderSpecificationSyntax Specification => lazySpecification.Value;
 
         public ObjectSyntax? Config => (this.WithClause as ProviderWithClauseSyntax)?.Config as ObjectSyntax;
 
         public IdentifierSyntax? Alias => (this.AsClause as AliasAsClauseSyntax)?.Alias;
 
         public override TextSpan Span => TextSpan.Between(this.Keyword, TextSpan.LastNonNull(this.SpecificationString, this.WithClause, this.AsClause));
 
-        SyntaxBase IArtifactReferenceSyntax.SourceSyntax => SpecificationString;
+        public SyntaxBase SourceSyntax => SpecificationString;
 
         public override void Accept(ISyntaxVisitor visitor) => visitor.VisitProviderDeclarationSyntax(this);
 
         public ArtifactType GetArtifactType() => ArtifactType.Provider;
 
-        public SyntaxBase? Path => this.Specification.BicepRegistryAddress is { } registryAddress ? SyntaxFactory.CreateStringLiteral(registryAddress) : null;
+        // if the provider specification is inlined return a value otherwise return null
+        public SyntaxBase? Path => this.Specification is InlinedProviderSpecificationSyntax spec ? SyntaxFactory.CreateStringLiteral(spec.UnexpandedArtifactAddress) : null;
+
+        public ResultWithDiagnostic<string> ResolveArtifactPath(RootConfiguration config)
+        {
+            if (this.Specification is InlinedProviderSpecificationSyntax inlinedSpec)
+            {
+                return new(inlinedSpec.UnexpandedArtifactAddress);
+            }
+
+            if (this.Specification is LegacyProviderSpecificationSyntax legacySpec)
+            {
+                return new("builtin");
```

Comment:
Where is this return value used? Should we use `string?` and return `null` to indicate an absense of a value, instead of a magic string?

Reply:
we don't really expect this value to be used. Removing